modules/primo/src/Primo/Ting/ObjectMapper.php

```diff
@@ -0,0 +1,232 @@
+   *
+   * @param \Ting\TingObjectInterface|NULL $object
+   *   An existing object to map the data onto. If none is provided a new
+   *   TingObject instance is used.
+   *
+   * @return \Ting\TingObject|\Ting\TingObjectInterface
+   *   The mapped object.
+   */
+  public function map(TingObjectInterface $object = NULL) {
+    // Construct the object if it does not exist.
+    if (NULL === $object) {
+      $object = new TingObject();
+    }
+
+    // Populate the object. Simple Object properties we have a 1:1 counterpart
+    // to in the Primo object is mapped directly. Properties that requires more
+    // processing before they can be mapped are handled by instance load*()
+    // functions.
+    $object->setId($this->document->getRecordId());
+    $object->setSourceId($this->document->getSourceRecordId());
+    // TODO BBS-SAL: Implement isLocal() method.
+    $object->setOwnerId($this->document->getSourceId());
+    $object->setTitle($this->document->getTitle());
+    // Primo does not distinguish between short and default title lengths.
+    $object->setShortTitle($this->document->getTitle());
+    $object->setAbstract($this->document->getLocalDisplayField(7));
+    // TODO BBS-SAL: Implement getAge() method.
+    // TODO BBS-SAL: Implement getAudience() method.
+    $object->setContributors($this->document->getContributors());
+    $object->setCreatorsFormatDefault($this->mapCreators(TingObjectInterface::NAME_FORMAT_DEFAULT));
+    $object->setCreatorsFormatSurnameFirst($this->mapCreators(TingObjectInterface::NAME_FORMAT_SURNAME_FIRST));
+    $object->setDescription($this->document->getDescription());
+    $object->setExtent($this->document->getDisplayFormat());
+    // TODO BBS-SAL: Implement getGenere() method.
+    $object->setIsbn($this->mapIsbn());
+    $object->setLanguage($this->mapLanguage());
+    $object->setMaterialSource($this->document->getSource());
+    $object->setOnlineUrl($this->document->getOnlineUrl());
+    $object->setOnline(!empty($this->document->getOnlineUrl()));
+    $object->setPublisher($this->document->getPublisher());
+    $object->setSeriesDescription($this->mapSeriesDescription());
+    $object->setSource($this->mapSource());
+    $object->setSubjects($this->mapSubjects());
+    // TODO BBS-SAL: Implement getTracks() method.
+    // TODO BBS-SAL: Implement getURI() method.
+    $object->setType($this->document->getType());
+    $object->setYear($this->document->getYear());
+
+
+    // Below are parts of the TingObjectInterface which the Primo modules
+    // currently do not support.
+    // Note that this does not necessarily mean that the information is not
+    // available from Primo. It is just not implemented at the moment. Please
+    // check each setter-invocation for any additional information.
+    // We do not support classification.
+    $object->setClassifications(FALSE);
+    // The contents of this field is partly duplicated by getExtent().
```

Comment:
Since we now have all this concentrated in a single code block just after the generic comment perhaps all these comments are not necessary except for cases were we "know" it is not supported e.g. `version`.